wini2c/i2capi.cpp

```diff
@@ -22,126 +24,194 @@ PyErr_SetString(PyExc_TypeError, errmsg); \
 return FAILURE; \
 }
 
-extern "C" {
-	void *new_i2cdevice(wchar_t *name, int slaveAddress, int busSpeed, int sharingMode) {
-		ComPtr<IInspectable> spInspectable = nullptr;
-        try {
-            I2cConnectionSettings^ settings = ref new I2cConnectionSettings(slaveAddress);
-            String^ deviceName = ref new String(name);
-            String^ querySyntax = I2cDevice::GetDeviceSelector(deviceName);
-			auto asyncop = DeviceInformation::FindAllAsync(querySyntax);
-			while (asyncop->Status != AsyncStatus::Completed) {
-				if (asyncop->Status == AsyncStatus::Error) {
-					PyErr_Format(PyExc_RuntimeError, "Could not find information for device '%S': %d", name, asyncop->ErrorCode);
-					return NULL;
-				}
-				Sleep(50);
-			}
-			auto info = asyncop->GetResults();
-			if (info == nullptr || info->Size == 0) {
-				PyErr_Format(PyExc_RuntimeError, "Could not find information for device '%S'", name);
-				return NULL;
-			}
-			String^ id = info->GetAt(0)->Id;
-
-            switch (busSpeed)
-            {
-            case FASTSPEED:
-                settings->BusSpeed = I2cBusSpeed::FastMode;
-                break;
-            case STANDARDSPEED:
-                settings->BusSpeed = I2cBusSpeed::StandardMode;
-                break;
-            default:
-                PyErr_Format(PyExc_TypeError, "Invalid I2C bus speed specified '%d'", busSpeed);
-                return NULL;
-            }
+static I2cDevice^ GetDeviceInboxDriver(int index, I2cConnectionSettings^ settings) {
+    if (index < 0) {
+        PyErr_Format(PyExc_RuntimeError, "Could not find i2c  id: '%d'", index);
+        return nullptr;
+    }
 
-            switch (sharingMode)
-            {
-            case SHAREDMODE:
-                settings->SharingMode = I2cSharingMode::Shared;
-                break;
-            case EXCLUSIVEMODE:
-                settings->SharingMode = I2cSharingMode::Exclusive;
-                break;
-            default:
-                PyErr_Format(PyExc_TypeError, "Invalid I2C sharing mode specified '%d'", sharingMode);
-                return NULL;
-            }
+    String^ deviceName = ref new String(L"I2C");
+    deviceName += (index + 1).ToString();
+    String^ querySyntax = I2cDevice::GetDeviceSelector(deviceName);
+    auto asyncop = DeviceInformation::FindAllAsync(querySyntax);
+    while (asyncop->Status != AsyncStatus::Completed) {
```

Comment:
That's Juanya's code.  I'm merely refactoring it.  My guess is that he doesn't want to deal with exceptions.